.azure-devops/tox.yml

```diff
@@ -0,0 +1,121 @@
+      'linters':
+        python: '3.7'
+        toxEnvStr: 'lint'
+        desc: 'Flake8, Pylint'
+      ${{ if parameters.python37 }}:
+        ${{ if parameters.azmin }}:
+          'python 3.7 tests, min CLI':
+            python: '3.7'
+            toxEnvStr: 'py3.7-azmin-unit'
+            desc: 'python 3.7 tests, min CLI'
+        ${{ if parameters.azcur }}:
+          'python 3.7 tests, current CLI':
+            python: '3.7'
+            toxEnvStr: 'py3.7-azcur-unit'
+            desc: 'Python 3.7 tests, current CLI'
+        ${{ if parameters.azdev }}:
+          'python 3.7 tests, dev CLI':
+            python: '3.7'
+            toxEnvStr: 'py3.7-azdev-unit'
+            desc: 'Python 3.7 tests, dev CLI'
+      ${{ if parameters.python38 }}:
+        ${{ if parameters.azmin }}:
+          'python 3.8 tests, min CLI':
+            python: '3.8'
+            toxEnvStr: 'py3.8-azmin-unit'
+            desc: 'python 3.8 tests, min CLI'
+        ${{ if parameters.azcur }}:
+          'python 3.8 tests, current CLI':
+            python: '3.8'
+            toxEnvStr: 'py3.8-azcur-unit'
+            desc: 'Python 3.8 tests, current CLI'
+        ${{ if parameters.azdev }}:
+          'python 3.8 tests, dev CLI':
+            python: '3.8'
+            toxEnvStr: 'py3.8-azdev-unit'
+            desc: 'Python 3.8 tests, dev CLI'
+      ${{ if parameters.python39 }}:
+        ${{ if parameters.azmin }}:
+          'python 3.9 tests, min CLI':
+            python: '3.9'
+            toxEnvStr: 'py3.9-azmin-unit'
+            desc: 'python 3.9 tests, min CLI'
+        ${{ if parameters.azcur }}:
+          'python 3.9 tests, current CLI':
+            python: '3.9'
+            toxEnvStr: 'py3.9-azcur-unit'
+            desc: 'Python 3.9 tests, current CLI'
+        ${{ if parameters.azdev }}:
+          'python 3.9 tests, dev CLI':
+            python: '3.9'
+            toxEnvStr: 'py3.9-azdev-unit'
+            desc: 'Python 3.9 tests, dev CLI'
+      ${{ if parameters.python310 }}:
+        ${{ if parameters.azmin }}:
+          'python 3.10 tests, min CLI':
+            python: '3.10'
+            toxEnvStr: 'py3.10-azmin-unit'
+            desc: 'Python 3.10 tests, min CLI'
+        ${{ if parameters.azcur }}:
+          'python 3.10 tests, current CLI':
+            python: '3.10'
+            toxEnvStr: 'py3.10-azcur-unit'
+            desc: 'Python 3.10 tests, current CLI'
+        ${{ if parameters.azdev }}:
+          'python 3.10 tests, dev CLI':
+            python: '3.10'
+            toxEnvStr: 'py3.10-azdev-unit'
+            desc: 'Python 3.10 tests, dev CLI'
+  steps:
+  - template: templates/clone-cli-repo.yml
+  - template: templates/setup-tox.yml
+  - template: templates/set-testenv-sentinel.yml
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: $(python)
+      architecture: 'x64'
+    displayName: 'Use cached Python $(python) for tests.'
+
+  - script: $(pyTools.pythonLocation)/bin/tox -e "$(toxEnvStr)" -vv
```

Comment:
What's the advantage of executing tox via `$(pyTools.pythonLocation)/bin/tox` ? Seems more OS specific. Could `python -m tox ....` invoke the same tox module in a more agnostic fashion?